snaps/promote-snaps.sh

```diff
@@ -7,18 +7,29 @@ set -eu
 # snaps/promote-snaps.sh
 
 SNAPS="kubectl kube-apiserver kube-controller-manager kube-scheduler kubelet kube-proxy cdk-addons kubeadm kubefed kubernetes-test"
+FAKE_PROMOTE="${FAKE_PROMOTE:-yes}"
 
 echo PROMOTE_FROM="$PROMOTE_FROM"
 echo PROMOTE_TO="$PROMOTE_TO"
 echo SNAPS="$SNAPS"
 
-echo "NOTE: Not actually releasing. Please look over this output carefully and run it manually.\n"
+. utils/retry.sh
+
+if [ "${FAKE_PROMOTE}" = "yes" ]
+then
+  echo "NOTE: Not actually releasing. Please look over this output carefully and run it manually.\n"
+fi
 
 for snap in $SNAPS; do
   revisions="$(snapcraft revisions $snap | grep " ${PROMOTE_FROM}\*" | cut -d " " -f 1)"
   for rev in $revisions; do
     for target in $PROMOTE_TO; do
-      echo snapcraft release $snap $rev $target
+      if [ "${FAKE_PROMOTE}" = "yes" ]
+      then
+        echo snapcraft release $snap $rev $target
+      else
+        retry snapcraft release $snap $rev $target
+      fi
```

Comment:
Is there any value in preserving the FAKE_PROMOTE=yes case? I'd rather just remove the `echo`s.

That was originally there just to play it safe when releasing to stable since I wasn't confident this script would do the right thing. We've used this script for at least four releases now and haven't seen any issues. I think we're good to just promote directly in all cases.

Reply:
+1